### PR TITLE
chore: upgrade Go to 1.17

### PR DESCRIPTION
**Pull request by davidby-influx**
Upgrade Go to 1.17

- [X] [Well-formatted commit messages](https://www.conventionalcommits.org/en/v1.0.0-beta.3/)
- [X] Rebased/mergeable
- [X] Tests pass

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,7 @@ parameters:
     # when updating the go version, should also update the go version in go.mod
     description: docker tag for cross build container from quay.io . Created by https://github.com/influxdata/edge/tree/master/dockerfiles/cross-builder .
     type: string
-    default: go1.15.10-f7b4e805fa9588c1c2fa4562ea29e576557fb797
+    default: quay.io/influxdb/cross-builder:go1.17.2-c9371c24dc3e9daabf19786cf9638428dfe2637a
```
Comment (lesam) [inline .circleci/config.yml:9]: To match the (newer) version 2.x is currently on

File: .circleci/config.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ parameters:
     # when updating the go version, should also update the go version in go.mod
     description: docker tag for cross build container from quay.io . Created by https://github.com/influxdata/edge/tree/master/dockerfiles/cross-builder .
     type: string
-    default: go1.15.10-f7b4e805fa9588c1c2fa4562ea29e576557fb797
+    default: go1.17.2-c9371c24dc3e9daabf19786cf9638428dfe2637a
```
Comment (lesam) [inline .circleci/config.yml:9]: To match the (newer) version 2.x is currently on

**Review by DStrand1 — DISMISSED** [review]
LGTM, that version of the cross-builder should have everything we need for the updated `protoc`

File: models/inline_strconv_parse.go [hunk 3]
```diff
@@ -30,15 +29,6 @@ func parseBoolBytes(b []byte) (bool, error) {
...
-// that require strings.
 func unsafeBytesToString(in []byte) string {
-	src := *(*reflect.SliceHeader)(unsafe.Pointer(&in))
-	dst := reflect.StringHeader{
-		Data: src.Data,
-		Len:  src.Len,
-	}
-	s := *(*string)(unsafe.Pointer(&dst))
-	return s
+	return *(*string)(unsafe.Pointer(&in))
```
Comment (lesam) [inline models/inline_strconv_parse.go:33]: `go vet` complained about this.

Use the same code as 2.x for this: https://github.com/influxdata/influxdb/blob/master/models/inline_strconv_parse.go#L31-L34

Comment (lesam) [reply models/inline_strconv_parse.go:33]: The trick is that (currently) the internal representation of string is the same as slice, except slice has an extra 'capacity'. So if you bitwise-interpret a byte slice as a string, it 'just works'.

**Review by DStrand1 — APPROVED** [review]

**Pull request merged**